message/messaging_service.cc

```diff
@@ -284,6 +284,35 @@ bool messaging_service::is_same_rack(inet_address addr) const {
     return topo.get_rack(addr) == topo.get_rack();
 }
 
+future<> messaging_service::ban_host(locator::host_id id) {
+    return container().invoke_on_all([id] (messaging_service& ms) -> future<> {
+        if (ms._banned_hosts.contains(id) || ms.is_shutting_down()) {
+            co_return;
+        }
+
+        auto [start, end] = ms._host_connections.equal_range(id);
+        std::vector<shared_ptr<rpc::server::connection>> conns;
+        conns.reserve(std::distance(start, end));
+        for (auto it = start; it != end; ++it) {
+            auto conn = it->second.server.get_connection(it->second.conn_id);
+            if (conn) {
+                conns.push_back(std::move(conn));
+            }
+        }
+
+        ms._banned_hosts.insert(id);
+        ms._host_connections.erase(start, end);
+
+        co_await parallel_for_each(conns, [] (shared_ptr<rpc::server_connection>& conn) {
+            return conn->stop();
```

Comment:
We could, but if some service is sending out RPCs to that node, `get_rpc_client` would just reestablish the connection.

I guess we could also ban outgoing connections like we do with incoming (and checking whether a host is banned in `get_rpc_client` or something).

Alternatively - and this is the approach I took basically - assume that once a node has left, *we* will not attempt to communicate with it (fingers crossed). This depends on nice behavior of our services. So we don't have to do the firewall in this direction, because we're not sending anything to that node anymore anyway.